Mathlib/Data/Set/Finite.lean

```diff
@@ -1207,6 +1207,12 @@ theorem Finite.dinduction_on {C : ∀ s : Set α, s.Finite → Prop} (s : Set α
   this h
 #align set.finite.dinduction_on Set.Finite.dinduction_on
 
+theorem Finite.induction_univ [Finite α] (C : Set α → Prop) (S0 : Set α)
+    (H0 : C S0) (ih : ∀ S ≠ Set.univ, C S → ∃ a ∉ S, C (insert a S)) : C Set.univ := by
+  refine' Finite.to_wellFoundedGT.wf.induction_bot' (fun S hS hS' ↦ _) H0
+  obtain ⟨a, ha, ha'⟩ := ih S hS hS'
+  exact ⟨_, Set.ssubset_insert ha, ha'⟩
```

Comment:
I guess there are several natural statements. E.g., you could also ask `\forall S, C S -> \forall a, C (insert a S)`.
Can you comment a bit on why you chose this statement? Should we just add both variants, and maybe even a third?

Also: should this be generalized to proving `C U` for a `U : Set X` with `hU: U.Finite`. And then drop the finiteness assumption on the ambient type.

Reply:
We already have the more standard `∀ a` version (it's just above, called `Finite.induction_on'`). My motivation for the `∃ a` version is that it's needed to prove that a transitive subgroup generated by transpositions must be the whole symmetric group.

I added the generalization you mentioned, although it's proof is currently a bit painful.

Reply:
I recall that the "bubble sort" induction PR https://github.com/leanprover-community/mathlib/pull/16536 was also aimed at symmetric groups ...